src/renderer/account/components/AccountPanel/Breakdown/Breakdown.js

```diff
@@ -28,61 +22,35 @@ export default class Breakdown extends React.PureComponent {
   };
 
   render() {
-    const data = this.getData();
+    const { className, balances, prices, currency } = this.props;
 
     return (
-      <ResponsiveContainer width="100%" className={classNames(styles.breakdown, this.props.className)}>
-        <PieChart>
-          <Pie
-            data={data}
-            dataKey="value"
-            nameKey="label"
-            innerRadius="50%"
-            outerRadius="80%"
-            startAngle={90}
-            endAngle={-270}
-            label={SectionLabel}
-            labelLine={false}
-            isAnimationActive={false}
-          >
-            {times(data.length, (index) => (
-              <Cell
-                key={`cell-${index}`}
-                fill={COLORS[index % COLORS.length]}
-                stroke={COLORS[index]}
-              />
-            ))}
-            <Label
-              id="totalValue"
-              position="center"
-              fontSize={28}
-              value={this.getTotalValue()}
-            />
-          </Pie>
-          <Tooltip formatter={this.formatValue} />
-        </PieChart>
-      </ResponsiveContainer>
+      <div className={classNames(styles.breakdown, className)}>
+        <TotalValue
+          className={styles.totalValue}
+          balances={balances}
+          prices={prices}
+          currency={currency}
+        />
+        <div className={styles.chart}>
+          <BreakdownChart
+            balances={balances}
+            prices={prices}
+            currency={currency}
+          />
+          {this.renderIcon()}
+        </div>
+      </div>
     );
   }
 
-  getData = () => {
-    const data = [...this.props.data];
+  renderIcon = () => {
+    const token = this.props.balances[0];
```

Comment:
is balances ordered or why is the first one always chosen?

Reply:
Yep, it gets sorted [right here](https://github.com/nos/client/pull/672/files#diff-5f5291c5fdf9f8bc0bee2fd66ad42f26R8).